### ISSUE TITLE
Assume utf-8 as default encoding

### ISSUE THREAD

**Issue by fbuchinger** [markusstraub/mdsplit#3]
Hi,

first of all thanks for this really useful script! Unfortunately I got a encoding errors when I ran it for the first time as Python 3.8 under windows assumes the cp1252 encoding on my computer. 
```Traceback (most recent call last):
  File "C:\dev\tmsspec-doc-py\mdsplit.py", line 341, in <module>
    main()
  File "C:\dev\tmsspec-doc-py\mdsplit.py", line 333, in main
    splitter.process()
  File "C:\dev\tmsspec-doc-py\mdsplit.py", line 147, in process
    self.process_file(self.in_path, self.out_path)
  File "C:\dev\tmsspec-doc-py\mdsplit.py", line 166, in process_file
    self.process_stream(stream, in_file_path.name, out_path)
  File "C:\dev\tmsspec-doc-py\mdsplit.py", line 88, in process_stream
    file.write(line)
  File "C:\Program Files\Python38\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\ue60f' in position 4: character maps to <undefined>`
```
Wouldn't make it sense if the script assumed  utf-8 encoding by default - maybe overrideable via CLI parameter? Had to monkey-patch the file.open() calls in your script with `encoding="utf-8"` to make it work.

Thanks,
Franz

**Comment by markusstraub** [comment]
Thanks for the bug report! My thinking was that if I don't specify the encoding in `open()`, the default encoding is used. However I guess the problem is that for Windows users the encoding of files may be UTF8 but may also be one of the codepages.

I think your system has Cp1252 set as default locale but then mdsplit breaks when it has to work wiht a UTF8 file?
If we always set the encoding to UTF-8 then the script will break if you want to work with a Cp1252 file.

I guess setting a predetermined encoding does not help much.. but maybe it solutions would be to
1) add an optional CLI parameter to force a specific encoding?
2) or to "simply" autodetect the encoding and use the same encoding for input and output

**Comment by fbuchinger** [comment]
I think an autodetection mechanism based on input encoding would be fine (encoding of first input file should determine target encoding), would add a fallback cli option if encoding detection fails.

**Comment by markusstraub** [comment]
I'll try https://pypi.org/project/cchardet/, but it will take a week or two.

**Comment by markusstraub** [comment]
I finally implemented a quite non-intrusive solution: a `--encoding` parameter that allows you to override the default encoding assumed by python. Not as fancy as using cchardet but simpler.

**Comment by markusstraub** [comment]
just released version 0.4.1
